test/unit/AppFooterController.js

```diff
@@ -0,0 +1,37 @@
+describe('AppFooterController', function() {
+  var $controller, $scope, service, serviceFlag;
+
+  beforeEach(module('myApp.controllers'));
+
+  beforeEach(function () {
+    serviceFlag = false;
+    service = {
+      switchLayout: function(parameter){
+        serviceFlag = true;
+      }
+    }
+
+    inject(function (_$controller_, _$rootScope_) {
+        $controller = _$controller_;
+
+        $scope = _$rootScope_.$new();
+        $controller('AppFooterController', {
+            $scope: $scope,
+            LayoutSwitchService: service
+        });
+    });
+  });
+
+  // define tests
+  it('compiles', function (done) {
+    expect(true).toBe(true);
+    done();
+  });
+
+  it('calls the right function', function (done) {
+    expect(serviceFlag).toBe(false);
+    $scope.switchLayout(null);
```

Comment:
The switchLayout function calls the switchLayout function of the LayoutSwitchService. The only concern of the test is to check if this function is called.
If the switchLayout function of the LayoutSwitchService is a toggle function, then that should be tested in the tests for the LayoutSwitchService
(for reference: controller.js, line 2980)